autotest/gdrivers/tiledb_write.py

```diff
@@ -40,34 +40,44 @@ def test_tiledb_write_complex():
 
     src_ds = gdal.Open('../gcore/data/cfloat64.tif')
 
-    new_ds = gdaltest.tiledb_drv.CreateCopy('tmp/tiledb_complex64', src_ds)
+    for mode in [ 'BAND', 'PIXEL', 'ATTRIBUTES' ]:
+        options = [
+            'INTERLEAVE=%s' % (mode)
+        ]
 
-    bnd = new_ds.GetRasterBand(1)
-    assert bnd.Checksum() == 5028, 'Didnt get expected checksum on still-open file'
+        new_ds = gdaltest.tiledb_drv.CreateCopy('tmp/tiledb_complex64', src_ds, options=options)
+        meta = new_ds.GetMetadata('IMAGE_STRUCTURE')
+        assert meta['INTERLEAVE'] == mode, 'Didnt get expected mode'
 
-    bnd = None
-    new_ds = None
-    gdaltest.tiledb_drv.Delete('tmp/tiledb_complex64')
+        bnd = new_ds.GetRasterBand(1)
+        assert bnd.Checksum() == 5028, 'Didnt get expected checksum on still-open file'
+
+        bnd = None
+        new_ds = None
+        gdaltest.tiledb_drv.Delete('tmp/tiledb_complex64')
 
 @pytest.mark.require_driver('TileDB')
 def test_tiledb_write_custom_blocksize():
     gdaltest.tiledb_drv = gdal.GetDriverByName('TileDB')
 
     src_ds = gdal.Open('../gcore/data/utmsmall.tif')
 
-    options = ['BLOCKXSIZE=32', 'BLOCKYSIZE=32']
+    for mode in [ 'BAND', 'PIXEL', 'ATTRIBUTES']:
```

Comment:
a more pytest'ic way would be to use pytest.parametrize (see autotest/gdrivers/fits.py for an example)